overlord/snapstate/snapstate.go

```diff
@@ -356,10 +356,16 @@ func CurrentInfo(s *state.State, name string) (*snap.Info, error) {
 	if err != nil && err != state.ErrNoState {
 		return nil, err
 	}
-	if sideInfo := snapst.CurrentSideInfo(); sideInfo != nil {
-		return readInfo(name, sideInfo)
+	return CurrentInfoWithSnapState(name, &snapst)
+}
+
+// CurrentInfoWithSnapState returns the information about the current revision of a snap with the given name and SnapState
+func CurrentInfoWithSnapState(name string, snapst *SnapState) (*snap.Info, error) {
+	sideInfo := snapst.currentSideInfo()
+	if sideInfo == nil {
+		return nil, &snap.NotFoundError{Snap: name}
```

Comment:
this error will mean broken no? we don't want to reuse it here I think (this will happen for something that has only a Candidate as well)